test/unit/memcached_test.rb

```diff
@@ -471,6 +471,32 @@ def test_get_multi_coerces_string_type
     end
   end
 
+  def test_get_multi_with_unresponsive_server
+    cache = Memcached.new(
+      [@servers.first, "invalid:43042"],
+    )
+
+    cache.set("bar", "bar")  # resolves to the first, valid server
+
+    # query both servers, and ensure order doesn't matter
+    get_res1 = cache.get(["foo", "bar"])
+    assert_equal({"bar" => "bar"}, get_res1)
+
+    get_res2 = cache.get(["bar", "foo"])
+    assert_equal({"bar" => "bar"}, get_res2)
```

Comment:
I want to test this under more failure modes, but preliminary data suggests that it's possible that if the last key raises a network error, `memcached_mget` will return that error even though it's possible to loop through fetching values for the servers that were able to respond. That's my current hypothesis, at least!